### PR TITLE
V2 upload answer

### PR DESCRIPTION
**Pull request by mbsimonovic**
- adds a new endpoint for answer upload
- adds a default error handler

### CODE REVIEW

**Comment by aaronshiel** [comment]
From reading the [docs on flask error handling](https://flask.palletsprojects.com/en/2.0.x/errorhandling/), the best practice for is to create error handles for each specific error type we want to handle, i.e. HTTPException and ValidationError, so I added handlers for those 2 cases. I also added a generic handler for any other Exception, but it's not very thorough in its reporting since Exception catches all sorts of errors. Any error type that we want handled in a specific way will need its own handler.

**Review by aaronshiel — APPROVED** [review]

File: mentor_upload_api/src/mentor_upload_api/__init__.py [hunk 1]
```diff
@@ -147,11 +152,20 @@ def create_app():
...
 
-    def error_handler(e):
-        # pass through HTTP errors
-        if isinstance(e, HTTPException):
-            return e
+    def generic_exception_handler(e):
+        """Return JSON instead of generic 500 internal error for Exceptions"""
+        response = app.response_class(
+            response=json.dumps({"error": "Exception", "message": str(e)}),
+            status=400,
```
Comment (mbsimonovic) [inline mentor_upload_api/src/mentor_upload_api/__init__.py:159]: this should return 500 right?

File: mentor_upload_api/src/mentor_upload_api/helpers.py [hunk 2]
```diff
@@ -31,9 +30,9 @@ def exec_graphql_with_json_validation(request_query, json_schema, **req_kwargs):
 def validate_json(json_data, json_schema):
     try:
         validate(instance=json_data, schema=json_schema)
-    except jsonschema.exceptions.ValidationError as err:
+    except ValidationError as err:
         logging.error(msg=err)
-        raise Exception(err)
+        raise ValidationError(err)
```
Comment (mbsimonovic) [inline mentor_upload_api/src/mentor_upload_api/helpers.py:35]:
```
        raise err
```

File: mentor_upload_api/src/mentor_upload_api/helpers.py [hunk 3]
```diff
@@ -31,9 +30,9 @@ def exec_graphql_with_json_validation(request_query, json_schema, **req_kwargs):
 def validate_json(json_data, json_schema):
     try:
         validate(instance=json_data, schema=json_schema)
-    except jsonschema.exceptions.ValidationError as err:
+    except ValidationError as err:
         logging.error(msg=err)
```
Comment (mbsimonovic) [inline mentor_upload_api/src/mentor_upload_api/helpers.py:34]: this should probably just be

```
        loggig.error(error)
```

File: mentor_upload_api/src/mentor_upload_api/helpers.py [hunk 4]
```diff
@@ -52,9 +51,9 @@ def json_validated_function(*args, **kwargs):
                     raise Exception("missing required param body")
                 validate(instance=json_body, schema=json_schema)
                 return f(json_body, *args, **kwargs)
-            except jsonschema.exceptions.ValidationError as err:
+            except ValidationError as err:
                 logging.error(msg=err)
-                raise Exception(err)
+                raise ValidationError(err)
```
Comment (mbsimonovic) [inline mentor_upload_api/src/mentor_upload_api/helpers.py:56]:
```
        raise err
```

**Pull request merged**